### PR TITLE
 Update protocol module to enable only  if input vpc_protocol_private_subnets_cidr_blocks exists

### PR DESCRIPTION
**Pull request by praveenramuu**
Update protocol module to enable only if input vpc_protocol_private_subnets_cidr_blocks exists

### CODE REVIEW

File: aws_scale_templates/sub_modules/vpc_template/main.tf [hunk 1]
```diff
@@ -182,7 +182,7 @@ module "compute_private_subnet" {
 # One protocol private subnet per provided AZ.
 module "protocol_private_subnet" {
   source       = "../../../resources/aws/network/subnet"
-  turn_on      = (var.cluster_type == "Storage-only" || var.cluster_type == "Combined-compute-storage") ? true : false
+  turn_on      = ((var.cluster_type == "Storage-only" || var.cluster_type == "Combined-compute-storage") && (var.vpc_protocol_private_subnets_cidr_blocks != null)) ? true : false
```
Comment (sasikeda) [inline aws_scale_templates/sub_modules/vpc_template/main.tf]: remove the extra brackets

File: aws_scale_templates/sub_modules/vpc_template/main.tf [hunk 2]
```diff
@@ -304,7 +304,7 @@ module "compute_private_route_table_association" {
 # Associate each protocol private subnet to one private route table.
 module "protocol_private_route_table_association" {
   source             = "../../../resources/aws/network/route_table_association"
-  turn_on            = (var.cluster_type == "Storage-only" || var.cluster_type == "Combined-compute-storage") ? true : false
+  turn_on            = ((var.cluster_type == "Storage-only" || var.cluster_type == "Combined-compute-storage") && (var.vpc_protocol_private_subnets_cidr_blocks != null)) ? true : false
```
Comment (sasikeda) [inline aws_scale_templates/sub_modules/vpc_template/main.tf]: remove the extra brackets

**Review by sasikeda — CHANGES_REQUESTED** [review]

File: aws_scale_templates/sub_modules/vpc_template/main.tf (hunk 2 above)
Comment (sasikeda) [reply aws_scale_templates/sub_modules/vpc_template/main.tf]: (var.cluster_type == "Storage-only" || var.cluster_type == "Combined-compute-storage") && var.vpc_protocol_private_subnets_cidr_blocks != null ? true : false

Use similar to this

**Comment by praveenramuu** [comment]
- Updated as per review comments - removed extra brackets

**Pull request merged**